<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="1" total="2" passed="0" failed="1" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-10-25T07:39:45 IST" name="Default suite" finished-at="2024-10-25T07:39:53 IST" duration-ms="7839">
    <groups>
    </groups>
    <test started-at="2024-10-25T07:39:45 IST" name="Default test" finished-at="2024-10-25T07:39:53 IST" duration-ms="7839">
      <class name="piku_mau.CartCreation">
        <test-method is-config="true" signature="setBaseURI()[pri:0, instance:piku_mau.CartCreation@2c4d1ac]" started-at="2024-10-25T07:39:45 IST" name="setBaseURI" finished-at="2024-10-25T07:39:45 IST" duration-ms="635" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setBaseURI -->
        <test-method is-config="true" signature="beforeTestMesthod()[pri:0, instance:piku_mau.CartCreation@2c4d1ac]" started-at="2024-10-25T07:39:45 IST" name="beforeTestMesthod" finished-at="2024-10-25T07:39:46 IST" duration-ms="250" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeTestMesthod -->
        <test-method signature="cart_already_exists_for_user()[pri:0, instance:piku_mau.CartCreation@2c4d1ac]" started-at="2024-10-25T07:39:46 IST" name="cart_already_exists_for_user" finished-at="2024-10-25T07:39:52 IST" duration-ms="6779" status="FAIL">
          <exception class="java.lang.IllegalArgumentException">
            <message>
              <![CDATA[Unrecognized field "error" (class models.UserSignupResponse), not marked as ignorable (3 known properties: "data", "headers", "statusCode"])
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: models.UserSignupResponse["error"])]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.IllegalArgumentException: Unrecognized field "error" (class models.UserSignupResponse), not marked as ignorable (3 known properties: "data", "headers", "statusCode"])
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: models.UserSignupResponse["error"])
at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4394)
at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4325)
at utilities.ApiResponseDeserializer.deserializeResponse(ApiResponseDeserializer.java:31)
at clients.UserClient.authenticateUser(UserClient.java:82)
at clients.UserClient.CreateCart(UserClient.java:92)
at piku_mau.CartCreation.cart_already_exists_for_user(CartCreation.java:48)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
at org.testng.SuiteRunner.run(SuiteRunner.java:330)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
at org.testng.TestNG.runSuites(TestNG.java:1099)
at org.testng.TestNG.run(TestNG.java:1067)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "error" (class models.UserSignupResponse), not marked as ignorable (3 known properties: "data", "headers", "statusCode"])
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: models.UserSignupResponse["error"])
at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1127)
at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1989)
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1700)
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1678)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:319)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:176)
at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4389)
... 31 more
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.IllegalArgumentException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- cart_already_exists_for_user -->
        <test-method is-config="true" signature="afterTestMethod()[pri:0, instance:piku_mau.CartCreation@2c4d1ac]" started-at="2024-10-25T07:39:53 IST" name="afterTestMethod" finished-at="2024-10-25T07:39:53 IST" duration-ms="2" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterTestMethod -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:piku_mau.CartCreation@2c4d1ac]" started-at="2024-10-25T07:39:53 IST" name="tearDown" finished-at="2024-10-25T07:39:53 IST" duration-ms="2" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- piku_mau.CartCreation -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
